lib/ecto/enum.ex

```diff
@@ -102,9 +114,14 @@ defmodule Ecto.Enum do
     rescue
       _ in UndefinedFunctionError -> raise ArgumentError, "#{inspect schema} is not an Ecto schema"
     else
-      %{^field => {:parameterized, Ecto.Enum, %{values: values}}} -> values
-      %{^field => {_, {:parameterized, Ecto.Enum, %{values: values}}}} -> values
-      %{} -> raise ArgumentError, "#{field} is not an Ecto.Enum field"
+      %{^field => {:parameterized, Ecto.Enum, %{values: values}}} ->
+        (Keyword.keyword?(values) && Keyword.keys(values)) || values
```

Comment:
See note above about normalizing `values`.

Reply:
To be clear, this should always return the atom keys. :)

Reply:
Fixed in 8f0b227f5db0eb7347c4601cac4f8377b13abd66 I think :-)